pkg/driver/driver.go

```diff
@@ -61,40 +75,12 @@ func (c RetryConnector) Connect(ctx context.Context) (driver.Conn, error) {
 
 // Driver implements part of the driver.Connector interface.
 func (c RetryConnector) Driver() driver.Driver {
-	return c.driver
+	return c.SqlDriver
 }
 
-// Driver wraps a driver.Driver that also must implement driver.DriverContext with logging capabilities and provides our RetryConnector.
-type Driver struct {
-	ctxDriver
-	Logger *logging.Logger
-}
-
-// OpenConnector implements the DriverContext interface.
-func (d Driver) OpenConnector(name string) (driver.Connector, error) {
-	c, err := d.ctxDriver.OpenConnector(name)
-	if err != nil {
-		return nil, err
-	}
-
-	return &RetryConnector{
-		driver:    d,
-		Connector: c,
-	}, nil
-}
-
-// Register makes our database Driver available under the name "icingadb-*sql".
+// Register sets the default mysql logger to the given one.
 func Register(logger *logging.Logger) {
-	sql.Register(MySQL, &Driver{ctxDriver: &mysql.MySQLDriver{}, Logger: logger})
-	sql.Register(PostgreSQL, &Driver{ctxDriver: PgSQLDriver{}, Logger: logger})
 	_ = mysql.SetLogger(mysqlLogger(func(v ...interface{}) { logger.Debug(v...) }))
```

Comment:
This could also be moved to the `mysql` case of `Database#Open()`. It's not an issue to call this method multiple times (with the same logger of course). Then we could move the constants and `RetryConnector` to the database package as well. The `driver` package really is empty now and we wouldn't have to do stuff like `icingaDriver` or `sqlDriver` and could remove `registerDriverOnce`.

Reply:
> hen we could move the constants and `RetryConnector` to the database package as well.

There's no such package in Icinga DB though! `Database#Open()` is located within the `config` package and moving the connector to that package feels out of place to me.

Reply:
True. I opt for renaming `driver` to `database` then. Our go libs will have that package and we still wouldn’t need that naming foo.

Reply:
`pkg/icingadb/` could also be an option as that's kind of the predecessor of `icinga-go-library/database`.

Both have the `DB` type (https://github.com/Icinga/icinga-go-library/blob/main/database/db.go, https://github.com/Icinga/icingadb/blob/main/pkg/icingadb/db.go) and moving this next to it doesn't feel out of place for me.